src/vs/editor/contrib/linesOperations/linesOperations.ts

```diff
@@ -947,6 +947,28 @@ export class LowerCaseAction extends AbstractCaseAction {
 	}
 }
 
+export class TitleCaseAction extends AbstractCaseAction {
+	constructor() {
+		super({
+			id: 'editor.action.transformToTitlecase',
+			label: nls.localize('editor.transformToTitlecase', "Transform to Title Case"),
+			alias: 'Transform to Title Case',
+			precondition: EditorContextKeys.writable
+		});
+	}
+
+	protected _modifyText(text: string): string {
+		return text.replace(/\S+/g, function (word: string) {
```

Comment:
Should we use `editor.wordSeparators` to split words?

Reply:
@rebornix well, however should we exclude single quote char?

For ex:

Excluded:
- `jaqra's commit` -> `Jaqra's Commit` ✔
- `this is 'single quote char'` -> `This Is 'single Quote Char'`❌

Included:
- `jaqra's commit` -> `Jaqra'S Commit`❌
- `this is 'single quote char'` -> `This Is 'Single Quote Char'`✔

My brain is burning 🧠🔥

Reply:
I think we should include (if it exists in word separators)

Reply:
Is that change acceptable on **`_modifyText` method's parameter**?

Here is some results:

![Capture](https://user-images.githubusercontent.com/48099350/55194965-22240c00-51bc-11e9-8dd6-deb5ce243299.JPG)